### ISSUE TITLE
Regression for `UnneededPercentQ`?

### ISSUE THREAD

**Issue by backus** [bbatsov/rubocop#2434]
Example:

```sh
$ rubocop -v
0.35.1
$ cat ex.rb .rubocop.yml
# encoding: utf-8

interpolation = 'hello'

%Q(example "with quotes" and #{interpolation})
---
Style/BarePercentLiterals:
  EnforcedStyle: percent_q
  SupportedStyles:
  - percent_q
  - bare_percent
$ rubocop ex.rb
Inspecting 1 file
C

Offenses:

ex.rb:5:1: C: Use %Q only for strings that contain both single quotes and double quotes, or for dynamic strings that contain double quotes.
%Q(example "with quotes" and #{interpolation})
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected
```

Gist for files in question: https://gist.github.com/backus/f341a4b92e54fa376211

Bisecting this points to 0ad7df5 I think

**Comment by backus** [comment]
Making the following change (which was present before #2274) seems to fix the issue and also does not break any specs

```diff
diff --git i/lib/rubocop/cop/style/unneeded_percent_q.rb w/lib/rubocop/cop/style/unneeded_percent_q.rb
index 055f002..5bad397 100644
--- i/lib/rubocop/cop/style/unneeded_percent_q.rb
+++ w/lib/rubocop/cop/style/unneeded_percent_q.rb
@@ -16,7 +16,7 @@ module RuboCop
         STRING_INTERPOLATION_REGEXP = /#\{.+}/

         def on_dstr(node)
-          check(node)
+          check(node) unless node.loc.expression.source =~ /"/
         end

         def on_str(node)
```

**Comment by backus** [comment]
I thought I was checking against master but looks like I had an old revision :(.  Ignore me this was fixed in dc45530893ec65014ec032798329e2d3cfea353c.
